src/subscription_manager/managercli.py

```diff
@@ -500,6 +532,20 @@ def main(self, args=None):
 
         # do the work, catch most common errors here:
         try:
+            if self.proxy_hostname or \
+               self.proxy_port or \
+               self.proxy_user or \
```

Comment:
proxy user/auth shouldn't result in a change that would cause a socket timeout. ie, if the proxy
host/port worked before, changing user/pass should just cause a auth fail (quickly). 

Doing it on any proxy info change may be reasonable, but that's a lot of code in there.